src/lib/storage/dictionary_column/dictionary_encoder.hpp

```diff
@@ -92,18 +104,31 @@ class DictionaryEncoder : public ColumnEncoder<DictionaryEncoder> {
 
     auto encoded_attribute_vector = compress_vector(attribute_vector, vector_compression_type(), alloc, {max_value});
 
-    auto dictionary_sptr = std::allocate_shared<pmr_vector<T>>(alloc, std::move(dictionary));
+    auto dictionary_sptr = std::allocate_shared<ValueVector<T>>(alloc, std::move(dictionary));
     auto attribute_vector_sptr = std::shared_ptr<BaseCompressedVector>(std::move(encoded_attribute_vector));
     return std::allocate_shared<DictionaryColumn<T>>(alloc, dictionary_sptr, attribute_vector_sptr,
                                                      ValueID{null_value_id});
   }
 
  private:
   template <typename T>
-  static ValueID _get_value_id(const pmr_vector<T>& dictionary, const T& value) {
+  static ValueID _get_value_id(const ValueVector<T>& dictionary, const T& value) {
     return static_cast<ValueID>(
         std::distance(dictionary.cbegin(), std::lower_bound(dictionary.cbegin(), dictionary.cend(), value)));
   }
+
+  template <typename T>
+  size_t _calculate_fixed_string_length(const pmr_concurrent_vector<T>& values) const {
+    return 0;
```

Comment:
This should not have to exist. You can do an `std::is_same` somewhere before to make sure you only call `calculate_fixed_string_length` on std::string.